docs/topics/object-declarations.md

```diff
@@ -1,23 +1,37 @@
 [//]: # (title: Object expressions and declarations)
 
-Sometimes you need to create an object that is a slight modification of some class, without explicitly declaring a new
-subclass for it. Kotlin can handle this with _object expressions_ and _object declarations_.
+Sometimes you need to use singletons for shared resources,
+create factory methods to efficiently instantiate objects,
+or modify the behavior of an existing class without explicitly declaring a new subclass.
+Kotlin handles these needs with _object expressions_ and _object declarations_.
+
+Object expressions and object declarations are best used for scenarios when:
+
+* **Using singletons for shared resources:** You need to ensure that only one instance of a class exists throughout the application.
+For example, managing a database connection pool.
+* **Creating factory methods:** You need a convenient way to create instances efficiently.
+[Companion objects](#companion-objects) allow you to define class-level methods and properties tied to a class, simplifying the creation and management of these instances.
+* **Modifying existing class behavior temporarily:** You want to modify the behavior of an existing class without creating a new subclass.
+For example, adding temporary functionality to an object for a specific operation.
+* **Limited class inheritance is desired:** You need to create single instances of classes without creating subclasses,
+such as for event listeners or callbacks. This ensures only one instance exists throughout the application.
```

Comment:
Sorry, for not leaving this comment earlier. 

I propose to drop this bulletpoint. It's technically incorrect "You need to create single instances of classes without creating subclasses" (anonymous subclass is technically still a new subclass, so we don't limit subclassing)

Other two bulletpoints "Modifying existing class behavior temporarily" and "Type-safe design is required" cover this bulletpoint good enough

Up to you

Reply:
thank you! 👍 I removed it